flink-yarn/src/main/scala/org/apache/flink/yarn/YarnJobManager.scala

```diff
@@ -754,6 +538,243 @@ class YarnJobManager(
       memoryLimit
     }
   }
+
+  /**
+    * Heartbeats with the resource manager and handles container updates.
+    */
+  object AMRMClientAsyncHandler extends AMRMClientAsync.CallbackHandler {
+
+    private var client : AMRMClientAsync[ContainerRequest] = null
+
+    override def onError(e: Throwable): Unit = {
+      self ! decorateMessage(
+        StopYarnSession(
+          FinalApplicationStatus.FAILED,
+          "Error in communication with Yarn resource manager: " + e.getMessage)
+      )
+    }
+
+    override def getProgress: Float = {
+      runningContainers.toFloat / numTaskManagers
+    }
+
+    override def onShutdownRequest(): Unit = {
+    }
+
+    override def onNodesUpdated(updatedNodes: JavaList[NodeReport]): Unit = {
+    }
+
+    override def onContainersCompleted(statuses: JavaList[ContainerStatus]): Unit = {
+
+      // TODO change this
```

Comment:
I wanted to remove the debug message here because heartbeat messages are processed in the heartbeat thread and logging a heartbeat on a callback does not really make sense. Thank you.